### ISSUE TITLE
Normalization for Librispeech pre-training data

### ISSUE THREAD

**Issue by rtst777** [andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning#14]
Hi, thanks for your excellent work on this repository.

I am wondering if you performed any additional normalization on the Librispeech dataset before pre-training the model (e.g. Mockingjay)? 

The only normalization function I found is [this](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L76). But did you also perform other normalization, e.g. normalize to zero mean and unit variance per speaker?

**Comment by andi611** [comment]
Thank you for having an interest in our work.

Yes, for Mockingjay:
First, features are extracted with Librosa [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L177-L182).
Next, delta is applied [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L188-L191).
And finally, cmvn is also applied [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L198) (the zero mean and unit variance you've mentioned).

**Comment by rtst777** [comment]
> Thank you for having an interest in our work.
> 
> Yes, for Mockingjay:
> First, features are extracted with Librosa [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L177-L182).
> Next, delta is applied [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L188-L191).
> And finally, cmvn is also applied [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L198) (the zero mean and unit variance you've mentioned).

Thank you for your answer! May I ask why you used delta instead of delta_delta? Doesn't delta_delta (240 dim) provide more information than delta (160 dim) for Mockingjay to learn?

**Comment by andi611** [comment]
Good point! We haven't verified delta_delta.

In our recent experiment, we find that the choice of acoustic features have an effect on model performance (These exps will be released soon with our new paper TERA).
We are still exploring which features are the best for these reconstruction-based models.
One thing certain for now is, acoustic features are a parameter choice that researchers have to explore.

**Comment by rtst777** [comment]
> Good point! We haven't verified delta_delta.
> 
> In our recent experiment, we find that the choice of acoustic features have an effect on model performance (These exps will be released soon with our new paper TERA).
> We are still exploring which features are the best for these reconstruction-based models.
> One thing certain for now is, acoustic features are a parameter choice that researchers have to explore.

make sense. look forward to your new paper :)

**Comment by CharlieZhaoyl** [comment]
> > Thank you for having an interest in our work.
> > Yes, for Mockingjay:
> > First, features are extracted with Librosa [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L177-L182).
> > Next, delta is applied [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L188-L191).
> > And finally, cmvn is also applied [here](https://github.com/andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning/blob/802e44c210b8fd0b9c5ab77113b835297385fcdb/utility/audio.py#L198) (the zero mean and unit variance you've mentioned).
> 
> Thank you for your answer! May I ask why you used delta instead of delta_delta? Doesn't delta_delta (240 dim) provide more information than delta (160 dim) for Mockingjay to learn?

Hi, thanks for the impressive work on this repository!

I have a question: Have you tried to simply concatenate different acoustic features like Mel, mfcc, fmllr, etc together to form the new input feature?  Because in intuition, I think features with higher dimensions may reach better results. I wonder whether this method make sense. Thanks!

**Comment by andi611** [comment]
> 
> 
> Hi, thanks for the impressive work on this repository!
> 
> 
> 
> I have a question: Have you tried to simply concatenate different acoustic features like Mel, mfcc, fmllr, etc together to form the new input feature?  Because in intuition, I think features with higher dimensions may reach better results. I wonder whether this method make sense. Thanks!
> 
> 

Interesting thought! 
It makes sense to me, in our recent study we find input features have an large effect on reconstruction-based pre-trained models.
We’ve tried mfcc, fbank, fmllr separately, but we haven’t used them in combination yet.
